edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Scraper.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+
+    public class Scraper : IMetricsScraper, IDisposable
+    {
+        const string UrlPattern = @"[^/:]+://(?<host>[^/:]+)(:[^:]+)?$";
+        static readonly Regex UrlRegex = new Regex(UrlPattern, RegexOptions.Compiled);
+        readonly HttpClient httpClient;
+        readonly Lazy<IDictionary<string, string>> endpoints;
+        readonly ILogger logger;
+
+        public Scraper(IList<string> endpoints, ILogger logger)
+        {
+            this.httpClient = new HttpClient();
```

Comment:
FYI, I found this blog post, https://softwareengineering.stackexchange.com/questions/330364/should-we-create-a-new-single-instance-of-httpclient-for-all-requests.  It may not require to reuse HttpClient instance.

Reply:
HttpClient is used in our existing codebase, how does existing implementation do?

Reply:
It currently creates a new httpclient for each request and disposes it after.